File: tests/proth_test.exs
ExUnit.start()

defmodule ProthTheoremTest do
  use ExUnit.Case

  test "3 is a prime number" do
    assert ProthTheorem.proth(3) == True
  end
  test "5 is a prime number" do
    assert ProthTheorem.proth(5) == True
  end
  test "9 is a prime number" do
    assert ProthTheorem.proth(9) == False
  end
  test "13 is a prime number" do
    assert ProthTheorem.proth(13) == True
  end
  test "17 is a prime number" do
    assert ProthTheorem.proth(17) == True
  end
  test "25 is a prime number" do
    assert ProthTheorem.proth(25) == False
  end
  test "33 is a prime number" do
    assert ProthTheorem.proth(33) == False
  end
  test "41 is a prime number" do
    assert ProthTheorem.proth(41) == True
  end
  test "49 is a prime number" do
    assert ProthTheorem.proth(49) == False
  end
  test "57 is a prime number" do
    assert ProthTheorem.proth(57) == False
  end
  test "65 is a prime number" do
    assert ProthTheorem.proth(65) == False
  end
  test "81 is a prime number" do
    assert ProthTheorem.proth(81) == False
  end
  test "97 is a prime number" do
    assert ProthTheorem.proth(97) == True
  end
  test "113 is a prime number" do
    assert ProthTheorem.proth(113) == True
  end
  test "129 is a prime number" do
    assert ProthTheorem.proth(129) == False
  end
  test "145 is a prime number" do
    assert ProthTheorem.proth(145) == False
  end
  test "161 is a prime number" do
    assert ProthTheorem.proth(161) == False
  end
  test "177 is a prime number" do
    assert ProthTheorem.proth(177) == False
  end
  test "193 is a prime number" do
    assert ProthTheorem.proth(193) == True
  end
  test "209 is a prime number" do
    assert ProthTheorem.proth(209) == False
  end
  test "225 is a prime number" do
    assert ProthTheorem.proth(225) == False
  end
  test "241 is a prime number" do
    assert ProthTheorem.proth(241) == True
  end
  test "257 is a prime number" do
    assert ProthTheorem.proth(257) == True
  end
  test "289 is a prime number" do
    assert ProthTheorem.proth(289) == False
  end
  test "321 is a prime number" do
    assert ProthTheorem.proth(321) == False
  end
  test "353 is a prime number" do
    assert ProthTheorem.proth(353) == True
  end
  test "385 is a prime number" do
    assert ProthTheorem.proth(385) == False
  end
  test "417 is a prime number" do
    assert ProthTheorem.proth(417) == False
  end
  test "449 is a prime number" do
    assert ProthTheorem.proth(449) == True
  end
  test "481 is a prime number" do
    assert ProthTheorem.proth(481) == False
  end
  test "513 is a prime number" do
    assert ProthTheorem.proth(513) == False
  end
  test "545 is a prime number" do
    assert ProthTheorem.proth(545) == False
  end
  test "577 is a prime number" do
    assert ProthTheorem.proth(577) == True
  end
  test "609 is a prime number" do
    assert ProthTheorem.proth(609) == False
  end
  test "641 is a prime number" do
    assert ProthTheorem.proth(641) == True
  end
  test "673 is a prime number" do
    assert ProthTheorem.proth(673) == True
  end
  test "705 is a prime number" do
    assert ProthTheorem.proth(705) == False
  end
  test "737 is a prime number" do
    assert ProthTheorem.proth(737) == False
  end
  test "769 is a prime number" do
    assert ProthTheorem.proth(769) == True
  end
  test "801 is a prime number" do
    assert ProthTheorem.proth(801) == False
  end
end

File: tests/proth_test.go
package proth_theorem

import (
  "github.com/obsessedyouth/Research-Assistant-Crypto/solutions/proth_theorem"
  "github.com/stretchr/testify/assert"
  "testing"
  )

func TestProth3(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(3))
}

func TestProth5(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(5))
}

func TestProth9(t *testing.T){
  assert.Equal(t, false,proth_theorem. Proth(9))
}

func TestProth13(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(13))
}

func TestProth17(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(17))
}

func TestProth25(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(25))
}

func TestProth33(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(33))
}

func TestProth41(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(41))
}

func TestProth49(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(49))
}

func TestProth57(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(57))
}

func TestProth65(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(65))
}

func TestProth81(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(81))
}

func TestProth97(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(97))
}

func TestProth113(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(113))
}

func TestProth129(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(129))
}

func TestProth145(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(145))
}

func TestProth161(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(161))
}

func TestProth177(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(177))
}

func TestProth193(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(193))
}

func TestProth209(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(209))
}

func TestProth225(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(225))
}

func TestProth241(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(241))
}

func TestProth257(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(257))
}

func TestProth289(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(289))
}

func TestProth321(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(321))
}

func TestProth353(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(353))
}

func TestProth385(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(385))
}

func TestProth417(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(417))
}

func TestProth449(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(449))
}

func TestProth481(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(481))
}

func TestProth513(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(513))
}

func TestProth545(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(545))
}

func TestProth577(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(577))
}

func TestProth609(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(609))
}

func TestProth641(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(641))
}

func TestProth673(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(673))
}

func TestProth705(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(745))
}

func TestProth737(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(737))
}

func TestProth769(t *testing.T){
  assert.Equal(t, true, proth_theorem.Proth(769))
}

func TestProth801(t *testing.T){
  assert.Equal(t, false, proth_theorem.Proth(801))
}
